ShopperInsights/README.md

```diff
@@ -0,0 +1,11 @@
+# Shopper Insights Module
```

Comment:
We haven't previously had module-specific READMEs - I'm open to it if we think it's useful and are going to do the same on iOS. What do other think about this?

Reply:
I don't have a strong preference of where to put it, but I'd like to see a short description of each of the modules and what capabilities they provide to the merchant/user.

Reply:
I think if we want to move to module specific READMEs this should be done in a separate PR where it's added for all modules vs just one. Though I do feel like a README within a module would be generally difficult to find for merchants. We also typically rely on our public docs to communicate capabilities.

Reply:
> We also typically rely on our public docs to communicate capabilities.

Do you mind showing me an example?
While there's the overall capabilities of the SDK, there's probably a distinction we want to make where we let the merchants know what exactly they get by including a particular module.

Reply:
I can totally see a root level MODULES.md or something similar where we can briefly describe what the module does and also what one needs to do to create a new module.

Reply:
Here is an example of the PayPal public docs: https://developer.paypal.com/braintree/docs/guides/paypal/overview/android/v4. The overview handles the product overview, then we have the client side + feature specific docs for the client side SDKs as well (our team is responsible for writing these). Typically most features also have a server side integration guide (our team doesn't typically write these) as well as a testing/going live section.

I think we tried having a form of docs live in both the repo and publicly on PPCP but it often resulted in one of the two being out of date so never truly having a "source of truth" for our merchants.

Reply:
One option might be to have links on the root readme file to the different module's readme files. This would help with the visibility issue.

If the content on each of the module's readme file is small, it might make more sense in the root readme. This way a developer doesn't need to navigate to see what each module does.

Reply:
> One option might be to have links on the root readme file to the different module's readme files. This would help with the visibility issue.

We already do that. I probably should have called it out to make it more clear.

Reply:
We need to leverage the public dev docs where possible and avoid additional doc maintenance on our. We're too small and don't have the bandwidth.

If it's common or accepted practice to the Android dev community that there's a README for each individual module, we can include that file with a basic template that points folks to the public dev docs.

Otherwise, let's not adopt this for the SDK.